lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/AllocatedEvaluator.java

```diff
@@ -96,4 +96,16 @@ void submitContextAndTask(final Configuration contextConfiguration,
   void submitContextAndServiceAndTask(final Configuration contextConfiguration,
                                       final Configuration serviceConfiguration,
                                       final Configuration taskConfiguration);
+
+  /**
+   * Check whether this evaluator is in closing state.
+   * @return whether this evaluator is in closing state.
+   */
+  boolean isClosing();
+
+  /**
+   * Check whether this evaluator is in closed state.
+   * @return whether this evaluator is in closed state.
+   */
+  boolean isClosed();
```

Comment:
> Do you think that this is okay 

Given that the header comment for `AllocatedEvaluator` says 

> Represents an Evaluator that is allocated, but is not running yet

 that's probably not the place to put these methods. Just because they work on an  evaluator which is actually running.

> have better idea?

I'll have a look.